### DISCUSSION TITLE
PatchItemAsync does not support IfMatchEtag property

### DISCUSSION THREAD

**Discussion by KT-JuliaDias** [Azure/azure-cosmos-dotnet-v3#4882] [General]
Hello,

I'm using Cosmos SDK in my application and using ETag optimistic locking for Upserts.
I'll need to use the PatchItemAsync method, however, it currently does not support the ETag property.

Do you have any plans to add this feature for patches?

Thank you, and looking forward to your response.

**Comment by FabianMeiswinkel** [comment]
An ETag means you have read the document already - why not use replace (with etag precondition)? Replace would have lower RU-charge than Patch and supports etag preconditions. PatchItem really is a convenience API for applications attempting to update documents wihtout reading the document before. If you absolutely want to do this you can use patch filter predicates - see https://github.com/Azure/azure-cosmos-dotnet-v3/blob/9bb076420cd8e6439aa61d1fd0331e1ede70abf0/Microsoft.Azure.Cosmos/src/RequestOptions/PatchItemRequestOptions.cs#L16-L34

**Reply by KT-JuliaDias** [reply]
makes sense! Thank you :D
